src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private Node[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSizeOfTable();
+        Node<K, V> newNode = new Node<>(null, key, value);
+        if (table[getBucket(key)] == null) {
+            table[getBucket(key)] = newNode;
+        } else {
+            Node<K, V> currentNode = table[getBucket(key)];
+            while (currentNode != null) {
+                if (Objects.equals(currentNode.key, newNode.key)) {
+                    currentNode.value = newNode.value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+            currentNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        if (table[getBucket(key)] == null) {
+            return null;
+        } else {
```

Comment:
let's remove if-else
if your `firstNode` is null you just won't go into while loop and will return null in the end